internal/gengapic/gengapic.go

```diff
@@ -43,7 +43,106 @@ const (
 
 var headerParamRegexp = regexp.MustCompile(`{([_.a-z0-9]+)`)
 
-// Gen is the entry point for GAPIC generation via the protoc plugin.
+// ParseOptions takes a string and parses it into a struct defining
+// customizations on the target gapic surface.
+// Options are comma-separated key/value pairs which are in turn delimited with '='.
+// Valid options include:
+// * go-gapic-package (package and module naming info)
+// * sample-only (only checked for presence)
+// * gapic-service-config (filepath)
+// * grpc-service-config (filepath)
+// * module (name)
+// * release-level (one of 'alpha', 'beta', or empty)
+// * transport ('+' separated list of transport backends to generate)
+// The only required option is 'go-gapic-package'.
+//
+// Valid parameter example:
+// go-gapic-package=path/to/out;pkg,module=path,transport=rest+grpc,gapic-service-config=gapic_cfg.json,release-level=alpha
+//
+// It returns a pointer to a populated options if no errors were encountered while parsing.
+// If errors were encountered, it returns a nil pointer and the first error.
+func ParseOptions(parameter *string) (*options, error) {
```

Comment:
There is already  parseOptions method in options.go. Why was this added?

Reply:
Transcription/merge error on my part that I'm still not sure how it happened. Good eye.